modbus_tcp_board/analog_pin.go

```diff
@@ -25,41 +26,41 @@ func (*ModbusAnalogPin) Close(ctx context.Context) error {
 }
 
 // Read implements board.AnalogReader.
-func (r *ModbusAnalogPin) Read(ctx context.Context, extra map[string]interface{}) (int, error) {
+func (r *ModbusAnalogPin) Read(ctx context.Context, extra map[string]interface{}) (board.AnalogValue, error) {
 	r.mu.RLock()
 	defer r.mu.RUnlock()
 
 	switch r.dataType {
 	case "int16":
 		val, err := r.board.client.ReadInt16(r.offset, r.pinType)
-		return int(val), err
+		return board.AnalogValue{Value: int(val)}, err
```

Comment:
I don't see a way of determining the range and stepSize of the analog reader this is using, so still just returning the bit value.

Reply:
Discussion question: how typical is it to not be able to provide this info? Is there other info that could have been provided instead if we'd designed this feature differently (if the scope doc had said something else)? I wonder if it will cause confusion for other people implementing their own board components in the wild.

Reply:
i believe its not typical. the issue in modbus's case is we could be talking to any board and we have no way to ask/guess what the info should be. Other boards will have an answer from the datasheet, or (in the revpis case) be configurable